File: backend/app/schemas/cargas.py
from datetime import datetime

from pydantic import BaseModel


class ResumenCargaPlaguicidas(BaseModel):
    filas_cargadas: int
    anios: list[int]
    meses_nuevos: list[int]
    filas_ya_cargadas: int
    filas_descartadas: int
    filas_truncadas: int
    combos_porcentaje: int
    porcentajes_no_parseables: int
    filas_sin_agrupador: int
    nomenclatura_actualizada: bool
    claves_nomenclatura_nuevas: int | None = None
    estado: str


class ResumenCargaNutrientes(BaseModel):
    filas_cargadas: int
    anios: list[int]
    meses_nuevos: list[int]
    filas_ya_cargadas: int
    filas_truncadas: int
    filas_sin_agrupador: int
    agrupador_actualizado: bool
    claves_agrupador_nuevas: int | None = None
    estado: str


class AuditoriaCargaItem(BaseModel):
    carga_id: int
    tipo_carga: str
    nombre_archivo: str | None
    usuario: str | None
    fecha_carga: datetime
    filas_procesadas: int | None
    filas_con_excepcion: int | None
    estado: str
    mensaje_error: str | None


class PaginaAuditoriaCargas(BaseModel):
    total: int
    pagina: int
    tamano_pagina: int
    filas: list[AuditoriaCargaItem]

File: frontend/src/types/cargas.ts
export interface ResumenCargaPlaguicidas {
  filas_cargadas: number;
  anios: number[];
  filas_descartadas: number;
  filas_truncadas: number;
  combos_porcentaje: number;
  porcentajes_no_parseables: number;
  filas_sin_agrupador: number;
  nomenclatura_actualizada: boolean;
  claves_nomenclatura_nuevas: number | null;
  estado: "OK" | "ConExcepciones" | "Error";
}

export interface ResumenCargaNutrientes {
  filas_cargadas: number;
  anios: number[];
  filas_truncadas: number;
  filas_descartadas_sin_f: number;
  filas_sin_agrupador: number;
  agrupador_actualizado: boolean;
  claves_agrupador_nuevas: number | null;
  estado: "OK" | "ConExcepciones" | "Error";
}

export interface AuditoriaCargaItem {
  carga_id: number;
  tipo_carga: string;
  nombre_archivo: string | null;
  usuario: string | null;
  fecha_carga: string;
  filas_procesadas: number | null;
  filas_con_excepcion: number | null;
  estado: string;
  mensaje_error: string | null;
}

export interface PaginaAuditoriaCargas {
  total: number;
  pagina: number;
  tamano_pagina: number;
  filas: AuditoriaCargaItem[];
}
